include/utils/compare_types.h

```diff
@@ -177,4 +177,71 @@ SUPERTYPE(double, long double);
 
 } // namespace libMesh
 
+#ifdef LIBMESH_HAVE_METAPHYSICL
+#define LIBMESH_DUAL_NUMBER_COMPARE_TYPES
```

Comment:
I probably should have a `HAVE` in there, but the MOOSE PR is already in...and this should be temporary